arabot/core/patches.py

```diff
@@ -195,6 +195,13 @@ def top_perm_role(self: disnake.Member) -> disnake.Role:
     return next(filter(is_perm_role, reversed(self.roles)), self.guild.default_role)
 
 
+@property
+def presence_count(self: disnake.Guild) -> int:
+    if self.approximate_presence_count is not None:
+        return self.approximate_presence_count
+    return sum(m.status is not disnake.Status.offline for m in self.members)
```

Comment:
The `sum(...)` part will only work correctly if the guild has no more than 75 000 members, because otherwise the Discord gateway only provides to the client the members who are in voice channels and the bot user itself. Of course, querying the additional members is possible, but it may be time and memory consuming.

Additionally, I'm not sure if disnake provides a way for this, but I'd suggest using the always present `member_count` (or similar) field's value instead of the optional `approximate_presence_count` which is the result of an HTTP API call which may or may not be queried by disnake.
This stands as long as my assumption about `approximate_presence_count` being the same as `approximate_member_count` from the `GUILD_CREATE` gateway message is correct.

Reply:
`approximate_member_count` counts all members in guild.
`approximate_presence_count` counts member that are not offline (even invisible ones, thanks Discord).
I do, in fact, use `member_count` on line 151 of the first commit:
> ```py
> guild.approximate_member_count or guild.member_count
> ```
Here `approximate_member_count` is not `None` only when fetching a `GuildPreview`, since all `Guild`s are pulled from cache.
By the way, using `or` will make this code fail if `GuildPreview` shows 0 members, but I believe that cannot happen.

Reply:
As for the first statement, I'll reply on that later.

Reply:
My bad for mistaking presence with members.